### PR TITLE
Add Manebooru Support

### PR DESCRIPTION
**Pull request by cultpony**
This Pull Request adds support for the Manebooru ([manebooru.art](https://manebooru.art)) Site.

I've run it through a basic test to ensure that account management works as well as posting art has no issues. Manebooru uses the same backend as Derpibooru so most of the code is copied from there with changes to make it work against Manebooru.

### CODE REVIEW

**Comment by mvdicarlo** [comment]
I'll try to take a look at this in the next couple days.
I appreciate the PR :)

File: electron-app/src/server/websites/manebooru/manebooru.service.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+@Injectable()
+export class Manebooru extends Website {
+  BASE_URL: string = 'https://manebooru.art';
+  acceptsFiles: string[] = ['jpeg', 'jpg', 'png', 'svg', 'gif', 'webm'];
+  acceptsSourceUrls: boolean = true;
+  enableAdvertisement: boolean = false;
+  defaultDescriptionParser = PlaintextParser.parse;
+  usernameShortcuts: UsernameShortcut[] = [
+    {
+      key: 'db',
```
Comment (mvdicarlo) [inline electron-app/src/server/websites/manebooru/manebooru.service.ts:37]: rename key to not collide with derpibooru (db) to something like "mb"

**Review by mvdicarlo — COMMENTED** [review]
Looks about right. Just have that one suggestion to change.
Did you test that posting it worked on your end?

**Comment by cultpony** [comment]
I force pushed the key rename to keep it simple. I did test this by uploading a piece of art with tags and rating, since it's the same backend as Derpibooru there shouldn't be an issue.

**Review by mvdicarlo — APPROVED** [review]
Looks good. Thanks for the hard work :)

**Comment by cultpony** [comment]
Thanks for merging it (and the awesome work on your end overall), I'll probably (maybe) make a similar PR for the non-plus version of PostyBirb if I find the time to work on that.

**Pull request merged**